python/lsst/base/packages.py

```diff
@@ -207,27 +216,67 @@ def getEnvironmentPackages():
     return packages
 
 
+def getCondaPackages():
+    """Get products and their versions from the conda environment.
+
+    Returns
+    -------
+    packages : `dict`
+        Keys (type `str`) are product names; values (type `str`) are their
+        versions.
+
+    Notes
+    -----
+    Returns empty result if a conda environment is not in use or can not
+    be queried.
+    """
+
+    try:
+        import json
+        from conda.cli.python_api import Commands, run_command
+    except ImportError:
+        return {}
+
+    # Get the installed package list
+    versions_json = run_command(Commands.LIST, "--json")
+    packages = {pkg["name"]: pkg["version"] for pkg in json.loads(versions_json[0])}
+
+    # Try to work out the conda environment name and include it as a fake
+    # package. The "obvious" way of running "conda info --json" does give
+    # access to the active_prefix but takes about 2 seconds to run.
+    # The equivalent to the code above would be:
+    #    info_json = run_command(Commands.INFO, "--json")
+    # As a comporomise look for the env name in the path to the python
+    # executable
+    match = re.search(r"/envs/(.*?)/bin/", sys.executable)
```

Comment:
I have tried to see if there was a slightly more elegant way of doing this without much luck. This may be about as good as it gets, unfortunately. Conda does seem to know some information about the environment(s) was activated, but I wasn't able to find a much better way than this.